.github/workflows/validate.yml

```diff
@@ -0,0 +1,239 @@
+
+      - name: Test
+        run: |
+          ./bin/minetest --run-unittests
+          
+  # Some builds doesn't require freetype, ensure it compiled properly
+  clang_9_no_freetype:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install compiler
+        run: |
+          sudo add-apt-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-9 main"
+          sudo apt-get install clang-9 -qyy
+          source ./util/github/common.sh
+          install_linux_deps
+
+      - name: Build
+        run: |
+          mkdir cmakebuild
+          cd cmakebuild
+          cmake -DCMAKE_BUILD_TYPE=Debug -DRUN_IN_PLACE=TRUE -DENABLE_GETTEXT=TRUE -DBUILD_SERVER=TRUE -DENABLE_FREETYPE=0 ${CMAKE_FLAGS} ..
+          make -j2
+        env:
+          CMAKE_FLAGS: "-DCMAKE_C_COMPILER=clang-9 -DCMAKE_CXX_COMPILER=clang++"
+
+      - name: Test
+        run: |
+          ./bin/minetest --run-unittests
+
+  # Run tests with valgrind to verify memleaks on the tested code
+  clang_9_valgrind:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install compiler
+        run: |
+          sudo add-apt-repository "deb http://apt.llvm.org/bionic/ llvm-toolchain-bionic-9 main"
+          sudo apt-get install clang-9 valgrind -qyy
+          source ./util/github/common.sh
+          install_linux_deps
+
+      - name: Build
+        run: |
+          mkdir cmakebuild
+          cd cmakebuild
+          cmake -DCMAKE_BUILD_TYPE=Debug -DRUN_IN_PLACE=TRUE -DENABLE_GETTEXT=TRUE -DBUILD_SERVER=TRUE ${CMAKE_FLAGS} ..
+          make -j2
+        env:
+          CMAKE_FLAGS: "-DCMAKE_C_COMPILER=clang-9 -DCMAKE_CXX_COMPILER=clang++"
+
+      - name: Test
+        run: |
+          valgrind --leak-check=full --leak-check-heuristics=all --undef-value-errors=no --error-exitcode=9 ./bin/minetest --run-unittests
+
+  win32:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install compiler
+        run: |
+          sudo apt-get update -qyy
+          sudo apt-get install wget
+          wget http://minetest.kitsunemimi.pw/mingw-w64-i686_9.2.0_ubuntu18.04.tar.xz -O mingw.tar.xz
+          sudo tar -xaf mingw.tar.xz -C /usr
+      - name: Build
+        run: |
+          ./util/buildbot/buildwin32.sh cmakebuild
+
+  win64:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install compiler
+        run: |
+          sudo apt-get update -qyy
+          sudo apt-get install wget
+          wget http://minetest.kitsunemimi.pw/mingw-w64-x86_64_9.2.0_ubuntu18.04.tar.xz -O mingw.tar.xz
+          sed -e "s|%PREFIX%|x86_64-w64-mingw32|" -e "s|%ROOTPATH%|/usr/x86_64-w64-mingw32|" < ./util/github/toolchain_mingw.cmake.in > ./util/buildbot/toolchain_mingw64.cmake
+          sudo tar -xaf mingw.tar.xz -C /usr
+      - name: Build
+        run: |
+          ./util/buildbot/buildwin64.sh cmakebuild
```

Comment:
what about `EXISTING_MINETEST_DIR` and `NO_MINETEST_GAME`?

If you let the script download fresh code from github.com/minetest/minetest it won't actually test the changes.
This is also incorrect in the `.gitlab-ci.yml` by the way.

Reply:
i understand for the NO_MINETEST_GAME flag. I just read the EXISTING_MINETEST_DIR old behaviour but i don't understand the point here, the build runs just perfectly on the CI

Reply:
If I create a PR that introduces a compiler error on Windows the following will happen:
* the CI runs
* it clones the source from `github.com/minetest/minetest`
* the source is master is bug-free so the build completes successfully
* the CI completes successfully

This misses the entire point.

Reply:
ok i see, will take a look